src/app/form/form.component.ts

```diff
@@ -0,0 +1,18 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-form',
+  templateUrl: './form.component.html',
+  styleUrls: ['./form.component.css']
+})
+export class FormComponent {
+  name:String='';
+
+  onclick(){
+    this.name='';
+  }
+
+  submit(event:Event){
+    this.name=(event.target as HTMLInputElement).value;
```

Comment:
Why you are using an event here? it should be only assigning an empty string for the name property!

Reply:
okkkey all is done